### PR TITLE
feat(footer-indicator): add footer indicator component 

### PR DESCRIPTION
**Pull request by sahiga**
This pull request adds a new footer indicator component that displays a puzzle piece icon and a small amount of descriptive text. The footer indicator is primarily intended for use in a `NavSidebar` or `LeftSidebar`.

In a `LeftSidebar`:
buie-footer-indicator-in-left-sidebar

In a `NavSidebar`:
buie-footer-indicator-in-nav-sidebar

With overflowed text:
buie-footer-indicator-with-tooltip

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @sahiga has signed the CLA. Thanks for the pull request!

File: src/components/footer-indicator/FooterIndicator.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+        if (offsetWidth < scrollWidth) {
+            // eslint-disable-next-line react/no-did-mount-set-state
+            this.setState({
+                isTextOverflowed: true,
+            });
+        }
+    }
+
+    indicatorTextElement: ?HTMLElement;
```
Comment (JustinHoldstock) [inline src/components/footer-indicator/FooterIndicator.js]: Can you move this to above the constructor? Also, a small code doc for it

File: src/icons/general/IconPuzzlePiece.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import AccessibleSVG from '../accessible-svg';
+
+type Props = {
+    className?: string,
+    color?: string,
+    dimension?: number,
+    title?: string,
+};
+
+const IconPuzzlePiece = ({ className = '', color = '#AFAFAF', dimension = 14, title }: Props) => (
```
Comment (JustinHoldstock) [inline src/icons/general/IconPuzzlePiece.js]: Can you add `#AFAFAF` to variables.js, and then update here and IconSearch.js?

Comment (sahiga) [reply src/icons/general/IconPuzzlePiece.js]: Is there a naming convention for these variables?

Comment (jstoffan) [reply src/icons/general/IconPuzzlePiece.js]: It looks like it already exists as `silverChalice` 🍷

File: src/components/footer-indicator/FooterIndicator.scss [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    width: 170px;
+    z-index: 10;
+}
+
+.bdl-FooterIndicator-content {
+    display: flex;
+    justify-content: flex-start;
+
+    &:focus {
+        outline: none;
```
Comment (jstoffan) [inline src/components/footer-indicator/FooterIndicator.scss]: It looks like this element is a div without a tabindex. Is `outline: none` still needed?

File: src/components/footer-indicator/FooterIndicator.scss [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    }
+}
+
+.bdl-FooterIndicator-iconWrapper {
+    margin: 8px 9px 8px 10px;
```
Comment (jstoffan) [inline src/components/footer-indicator/FooterIndicator.scss]: Can we use flex alignment to vertically center these elements?

File: src/components/footer-indicator/FooterIndicator.scss [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+.bdl-FooterIndicator-text {
+    color: $white;
+    font-size: 11px;
+    margin: 6px 10px 6px 0;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+
+    &::after {
```
Comment (jstoffan) [inline src/components/footer-indicator/FooterIndicator.scss]: Is this needed? If so, a comment may be helpful.

Comment (sahiga) [reply src/components/footer-indicator/FooterIndicator.scss]: I originally added this as a workaround to hide tooltips for truncated content in Safari (https://stackoverflow.com/questions/20974276/prevent-safari-from-showing-tooltip-when-text-overflow-is-hidden-with-ellipsis). See LeftSidebarLink.scss for an example. However, I have just confirmed that we can show the tooltip regardless of the length of the text, so I will remove this rule 🙂

File: src/components/footer-indicator/FooterIndicator.scss (hunk 3 above)
Comment (sahiga) [reply src/components/footer-indicator/FooterIndicator.scss]: Yes. The fuzzy blue outline will still show up on focus, which would suggest that this element is clickable when it is not.

Comment (jstoffan) [reply src/components/footer-indicator/FooterIndicator.scss]: I see, thanks. It looks like `Tooltip` adds a `tabindex` attribute to its immediate child component.

**Comment by sahiga** [comment]
@JustinHoldstock @jstoffan @jeremypress Can you review this PR again? Thank you 🙏

File: src/features/left-sidebar/LeftSidebar.js [hunk 6]
```diff
@@ -18,6 +18,7 @@ import NewItemsIndicator from './NewItemsIndicator';
 import type { Callout } from './Callout';
 
 import './styles/LeftSidebar.scss';
+import FooterIndicator from '../../components/footer-indicator/FooterIndicator';
```
Comment (jstoffan) [inline src/features/left-sidebar/LeftSidebar.js]: Can you move this up to its friends so the SCSS import can be last?

Comment (sahiga) [reply src/features/left-sidebar/LeftSidebar.js]: @jstoffan Done!

**Pull request merged**